src/interactions/movethread.js

```diff
@@ -54,10 +57,29 @@ module.exports = {
         return utils.postInteractionError(interaction, `This thread is already in ${targetCategory.name}.`, null, true);
       }
       return threads.moveThread(thread, targetCategory, false)
-        .catch((e) => {
-          utils.handleError(e);
-          utils.postInteractionError(interaction, "Something went wrong while attempting to move this thread.", null, true);
-        });
+        .then(
+          () => utils.postInteractionSuccess(interaction, "***Thread has been moved to community***", null, false),
+          (e) => {
+            utils.handleError(e);
+            utils.postInteractionError(interaction, `Something went wrong. ${e.message}`, null, true);
+          }
+        );
+    } else if (customID === "support") {
+      const targetCategory = message.channel.guild.channels.get(config.supportThreadCategoryId);
+      if (! targetCategory) {
+        return utils.postInteractionError(interaction, `I can't move this thread to ${customID.toLowerCase()} because it doesn't exist.`);
+      }
+      if (message.channel.parentID === targetCategory.id) {
+        return utils.postInteractionError(interaction, `This thread is already in ${targetCategory.name}.`, null, true);
+      }
+      return threads.moveThread(thread, targetCategory, false)
+        .then(
+          () => utils.postInteractionSuccess(interaction, "***Thread has been moved to support***", null, false),
+          (e) => {
+            utils.handleError(e);
+            utils.postInteractionError(interaction, `Something went wrong. ${e.message}`, null, true);
+          }
+        );
```

Comment:
Is there any way we can remove the duplicate code here?

Reply:
Updated